ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -290,32 +303,41 @@ boolean isChunkFillComplete() {
    */
   void fillChunks()
       throws InterruptedException {
-    PutChunk chunkToFill;
-    if (!chunkFillingCompleted && !operationCompleted) {
-      do {
-        // Attempt to fill a chunk
-        if (channelReadBuffer == null) {
-          channelReadBuffer = chunkFillerChannel.getNextChunk(0);
-        }
-        if (channelReadBuffer != null) {
-          chunkToFill = getChunkToFill();
-          if (chunkToFill == null) {
-            // no chunks are free to be filled yet.
-            break;
+    try {
+      PutChunk chunkToFill;
+      if (!chunkFillingCompleted && !operationCompleted) {
+        do {
+          // Attempt to fill a chunk
+          if (channelReadBuffer == null) {
+            channelReadBuffer = chunkFillerChannel.getNextChunk(0);
           }
-          bytesFilledSoFar += chunkToFill.fillFrom(channelReadBuffer);
-          if (!channelReadBuffer.hasRemaining()) {
-            chunkFillerChannel.resolveOldestChunk(null);
-            channelReadBuffer = null;
+          if (channelReadBuffer != null) {
+            chunkToFill = getChunkToFill();
+            if (chunkToFill == null) {
+              // no chunks are free to be filled yet.
+              break;
+            }
+            bytesFilledSoFar += chunkToFill.fillFrom(channelReadBuffer);
+            if (!channelReadBuffer.hasRemaining()) {
+              chunkFillerChannel.resolveOldestChunk(null);
+              channelReadBuffer = null;
+            }
+          } else {
+            // channel does not have more data yet.
+            break;
           }
-        } else {
-          // channel does not have more data yet.
-          break;
+        } while (bytesFilledSoFar < blobSize);
+        if (bytesFilledSoFar == blobSize) {
+          chunkFillingCompleted = true;
         }
-      } while (bytesFilledSoFar < blobSize);
-      if (bytesFilledSoFar == blobSize) {
-        chunkFillingCompleted = true;
       }
+    } catch (InterruptedException e) {
+      throw e;
```

Comment:
My rationale was that `InterruptedException`s were bubbled up before and I wanted to maintain that behavior. If it is safe to stop only this operation, we can swallow it.